frontend/components/label/FormCreateDimension.vue

```diff
@@ -0,0 +1,515 @@
+  },
+
+  methods: {
+    async setAssignedDimensions() {
+      const base = this as any
+      await base.$services.dimension.list(base.projectId).then((response: any) => {
+        base.assignedDimensions = response.items.map((item: any) => {
+          const groupMap: any = {
+            DIM_OTH: 'Others',
+            DIM_OF: 'Offensive',
+            DIM_HUM: 'Humor',
+            DIM_EMO: 'Emotions'
+          }
+          if (item.metadata && item.metadata.length) {
+            const { codename } = item.metadata[0]
+            const groupMapKey: string =
+              Object.keys(groupMap).find((key) => codename.includes(key)) || ''
+            item.group = groupMap[groupMapKey] || 'Dynamic'
+          }
+          return item
+        })
+      })
+    },
+    async setDimensionList() {
+      const base = this as any
+      await base.$services.dimension.listAllDimensions().then((response: any) => {
+        base.allDimensions = response.items
+      })
+      await base.setAssignedDimensions()
+    },
+    getDimensionDetailPreviewComponent() {
+      const base = this as any
+      if (base.formData.dimensionType === 'slider') {
+        return SliderInput
+      } else if (base.formData.dimensionType === 'checkbox') {
+        return CheckboxInput
+      }
+    },
+    getDimensionDetailFormComponent() {
+      const base = this as any
+      if (base.formData.dimensionType === 'slider') {
+        return SliderForm
+      } else if (base.formData.dimensionType === 'checkbox') {
+        return CheckboxForm
+      }
+    },
+    resetForm() {
+      const base: any = this
+      const refForm: any = base.$refs.form
+      refForm && refForm.resetValidation()
+      base.formData = {
+        ...base.formData,
+        ...{
+          dimensions: [],
+          dimensionName: '',
+          dimensionType: '',
+          isCreatingNewDimension: true,
+          required: false,
+          readOnly: false,
+          slider: {
+            sliderMin: 0,
+            sliderMax: 10,
+            sliderStep: 1,
+            minValDescription: '',
+            maxValDescription: '',
+            withCheckbox: false,
+            checkboxCodename: ''
+          },
+          checkbox: {
+            isMultipleAnswers: false,
+            minAnswerNumber: 1,
+            maxAnswerNumber: 1,
+            options: []
+          }
+        }
+      }
+    },
+    getCreateDimensionFormRequests(): any {
+      const base = this as any
+      const dynamicGroups = base.assignedDimensions.filter((item: any) => item.group === 'Dynamic')
+      const requests = []
+      const request = {
+        name: base.formData.dimensionName,
+        type: base.formData.dimensionType,
+        dimension_meta_data: [
+          {
+            codename: `DIM_DYN_${dynamicGroups.length + 1}`,
+            config: {},
+            required: Number(base.formData.required),
+            readonly: Number(base.formData.readOnly)
+          }
+        ]
+      }
```

Comment:
Thanks for finding the bug, updated.

Reply:
Hello, I have tested again and unfortunately it hasn't been fixed. I see that we now use `base.allDimensions`, but this still fails because the items in `base.allDimensions` do not have the `.group` property (I think the `.group` property is only assigned to items in `base.assignedDimensions`).
<img width="631" alt="Capture" src="https://user-images.githubusercontent.com/64476430/228527251-fcec2e5f-050b-4534-a83c-beee6bf3c4e4.PNG">

A quick fix would be adding the following code to line 289 inside `setDimensionList()` - may require refactor because it is copied and adapted from the same mechanism in `setAssignedDimensions()`:
```
      base.allDimensions = base.allDimensions.map((item: any) => {
        const groupMap: any = {
          DIM_OTH: 'Others',
          DIM_OF: 'Offensive',
          DIM_HUM: 'Humor',
          DIM_EMO: 'Emotions'
        }
        if (item.metadata && item.metadata.length) {
          const { codename } = item.metadata[0]
          const groupMapKey: string =
            Object.keys(groupMap).find((key) => codename.includes(key)) || ''
          item.group = groupMap[groupMapKey] || 'Dynamic'
        }
        return item
      })
```